ADR/0033-provisioning-test-resources.md

```diff
@@ -0,0 +1,277 @@
+    end
+
+    subgraph hive [Hive Namespace]
+      HiveOperator
+    end
+
+    subgraph cluster-aas [ClaaS Namespace]
+      ClaaSOperator
+      ArgoCDOperator
+      ClusterTemplates
+      ApplicationSets
+    end
+  end
+
+  ClusterTemplateInstances --> |reference w/releaseImage| ClusterTemplates
+  ClusterTemplates --> |reference| ApplicationSets
+  ClusterDeployments ---> |reference| install-configs
+  ClusterDeployments ---> |reference| aws-creds
+  ClusterDeployments ---> |reference| cluster-ssh-key
+  ClusterDeployments ---> |reference| cluster-pull-secret
+  ClusterPools ---> |reference| install-configs
+  ClusterPools ---> |reference| aws-creds
+  ClusterPools ---> |reference| cluster-ssh-key
+  ClusterPools ---> |reference| cluster-pull-secret
+  ArgoCDOperator --> |installs| HelmCharts
+  ArgoCDOperator --> |watches| Applications
+  HiveOperator --> |watches| ClusterDeployments
+  HiveOperator --> |watches| ClusterClaims
+  HiveOperator --> |4. creates| kubeconfig
+  install-configs --> |contain copy of| cluster-pull-secret
+  install-configs --> |contain copy of| cluster-ssh-key
+  HelmCharts --> |3. deploy| ClusterDeployments
+  HelmCharts --> |3. deploy| ClusterClaims
+  ApplicationSets --> |reference| HelmCharts
+  ClusterClaims --> |reference| ClusterPools
+  IntegrationPipeline ---> |1. creates| ClusterTemplateInstances
+  IntegrationPipeline ---> |5. reads| kubeconfig
+  ClaaSOperator --> |2. creates| Applications
+  ClaaSOperator --> |watches| ClusterTemplateInstances
+```
+
+### Scalability
+
+The [CaaS] Operator should scale well. It hands off most workloads to ArgoCD. Even so,
+there are a few good reasons to deploy CaaS, Hive and/or Hypershift on clusters separate from the
+Konflux dataplane clusters:
+
+* Hive's upper limits, once reached, are ultimately only surmountable by adding more management
+  clusters.
+* The tight coupling of Hypershift management cluster version to provisionable control plane
+  versions suggests timely upgrades of the management cluster may be important to our end users.
+  The rest of the Konflux services on the dataplane may not support as aggressive of an upgrade
+  schedule.
+
+### Access Management
+  
+Introducing new cluster(s) creates complexity elsewhere. A tenant needs access to a namespace
+on the remote cluster within which they can manage select resources
+(e.g. `ClusterTemplateInstances`, `Secrets`, `ClusterPools`).
+
+We will either update or create [NSTemplateTiers] with the addition of a `SpaceRequest`. A new
+cluster role will be created on the `ToolchainCluster` Custom Resource to classify the cluster(s)
+used for test environment provisioning. The `SpaceRequest` controller, noticing the
+cluster role on the request, will create the namespace on one of the remote clusters. It
+will also create a secret in the tenant namespace containing credentials for a service account
+with access to the remote namespace. This secret can then be used from a `PipelineRun` workload
```

Comment:
Is there any concern if this secret is extracted out of the tenant workspace (i.e. by an Admin)?

Reply:
I think the concern should be pretty limited considering these are ephemeral clusters that will only live for a limited duration which we control. Such concerns would be applicable to any solution since the required deliverable to the user is a kubeconfig/cluster-admin creds accessing from within an integration PipelineRun.